pkg/controllers/provisioning/scheduling/node.go

```diff
@@ -102,3 +151,22 @@ func (n Node) hasCompatibleResources(resourceList v1.ResourceList, it cloudprovi
 	}
 	return true
 }
+
+func (n *Node) recalculateDaemonResources() {
+	n.daemonResources = nil
+	for _, daemon := range n.daemons {
+		podRequirements := v1alpha5.NewPodRequirements(daemon)
+		// we intentionally do not check if the DaemonSet pod will fit on the node here, we want large DS pods to cause
+		// us to choose larger instance types
+		if err := n.Requirements.Compatible(podRequirements); err != nil {
+			continue
+		}
+		if err := n.Taints.Tolerates(daemon); err != nil {
```

Comment:
Can we filter this beforehand to avoid recalculating on every add? Taints shouldn't change.

Reply:
Yup, adding that.